Add infinite loop to run parser script

gaisin opened the pull request:
Adding infinite loop to run parser script periodically using [`schedule`](https://schedule.readthedocs.io) package. Based on PR #4.

OdintsovTim commented on susp/parser.py:
```diff
@@ -248,7 +171,68 @@ def get_event_url(event):
+                LOG.debug("event was saved")
+                if is_available and not saved_event.is_available:
+                    LOG.debug("trying to send notification on saved event")
+                    susp.notifications.available_again(saved_event)
```
Информация о мероприятии после того как оно стало снова доступно может поменяться, в частности информаци о цене и оставшихся свободных мест. Лучше обновить эту информацию перед тем как посылать сообщение в канал

OdintsovTim replied:
event.is_available не меняй пожалуйста, мне это нужно будет для проверки новое это мероприятие, или старое, которое стало доступно

gaisin replied:
> Лучше обновить эту информацию перед тем как посылать сообщение в канал

Согласен, добавлю

> event.is_available не меняй пожалуйста, мне это нужно будет для проверки новое это мероприятие, или старое, которое стало доступно

По поводу `is_available` предлагаю не переносить в модуль `notifications` логику проверки обновления, а оставить ее здесь, а в функциях нотификации использовать только переданный инстанс класса `Event()`

gaisin committed:
parser: remove ipython notebook draft

gaisin committed:
docker: add Dockerfile for app container

gaisin committed:
requirements: add missing packages

gaisin committed:
event: change price field type from int to str and specify host for connection

Because some of the price values can contain letters.

gaisin committed:
parser: get rid of unnecessary comments and impove naming

gaisin committed:
parser: move main parsing function to parser.py

gaisin committed:
notifications: import settings as a susp module

gaisin committed:
utils: change logs path

gaisin committed:
app: add entry point for the parser

gaisin committed:
docker: add app container description

gaisin committed:
readme: add quick start guide and remove unnecessary guide for jupiter notebook

OdintsovTim commented on README.md:
````diff
@@ -1,4 +1,29 @@
+3. Add settings.py to `susp/` folder by the following template:
+    ```python
+    import telegram
+    TOKEN = 'asueyfajhsdfkjywerkj2634hg123kfdlfjalsdkfj'  # telegram bot token
````
Я бы написал TOKEN = 'your_bot_token' , чтобы не вводить людей в заблуждение

gaisin replied:
Согласен, поправил

OdintsovTim commented on susp/event.py:
```diff
@@ -19,7 +19,7 @@ class Event(Document):
 
     datetime_str = StringField(required=True, unique=True)
-    price = IntField()
+    price = StringField()
```
Почему цена строковая? Тогда лучше и количество мест сделать строковой, если там информация иногда подается как строка?

gaisin replied:
Количество мест тоже сделал строковой
> Почему цена строковая?

Потому что иногда в тексте цены могут появляться не только цифры, но и другие символы

OdintsovTim commented on susp/parser.py:
```diff
@@ -182,7 +105,7 @@ def get_event_poster_url(event):
 
 
-def is_available(event):
+def get_event_availability(event):
```
Подскажи пожалуйста, что делает эта функция?

gaisin replied:
Как и написано в докстринге, `Returns availability of given event.`)
По переданному ей спаршенному мероприятию отвечает на вопрос "На данное мероприятие можно купить билет?"

gaisin committed:
event: change seats_left field type from int to str

Also since type is str now, add new way to parse information about remaining tickets..
